controllers/cloud.redhat.com/clowdenvironment_controller.go

```diff
@@ -97,9 +103,14 @@ func (r *ClowdenvironmentReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		r.log.Error(err, "could not retrieve newly created namespace", "namespace", namespaceName)
 	}
 
-	if err := r.client.Update(ctx, &namespace); err != nil {
-		return ctrl.Result{}, err
-	}
+	err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		if err = r.client.Update(ctx, &namespace); err != nil {
+			return err
+		}
+
+		r.log.Info(fmt.Sprintf("Updated clowdenvironment status for namespace [%s]", namespaceName))
+		return err
+	})
```

Comment:
So, the logic needs some work:
* It's confusing using the same `err` error name for both the inner function and the outer retry logic.
* Line 112 should return `nil`, we already know there is no error because we checked it on line 107
* The error from line 106, is never checked - If the retry logic fails to successfully finish the function, 

**But!**

Reading the documentation around this, this will basically always result in a ConflictError being returned, so the only effect of running this will be to delay reconciliation even further. The reason for this is because of _what_ you are doing inside function. You are taking the same resource and just trying to apply it again. This won't work. There was a conflict, and so you need to run the entire reconciliation again to allow k8s to figure out if there is a conflict _now_. 

This being said, I'm not sure with the way that most of our operators are set to function, that it makes sense to use this. Running the entire reconciliation process again is easier left to, well, running the entire reconciliation process again. I don't think this will save you any time, and in the worst case scenario, will actually probably make things slower.

Sorry.